projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,234 @@
+package hasadna.noloan2;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Environment;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import com.itextpdf.licensekey.LicenseKey;
+import com.itextpdf.html2pdf.*;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import noloan.R;
+
+
+public class LawsuitPdfActivity extends AppCompatActivity {
+    // TODO: Add logs to class functions.
+    private static final String TAG = "LawsuitPdfActivity";
+
+    // Folder names
+    private static final String LAWSUIT_MAIN_DIR_NAME = "lawsuits";
+    private static final String LAWSUIT_TEMPLATE_DIR_NAME = "templates";
+    private static final String LAWSUIT_OUTPUT_DIR_NAME = "pdf";
+    private static final String LAWSUIT_TEMPLATE_FILE_NAME = "template.xhtml";
+
+    // Folder paths - initDirStructure()
+    private String lawsuitMainPath = "";
+    private String lawsuitTemplateDirPath = "";
+    private String lawsuitTemplateFilePath = "";
+    private String lawsuitOutputPath = "";
+
+    // Date formats
+    private static final String DATE_TIME_FORMAT = "dd-M-yyyy hh-mm-ss";
+    private static final String DATE_FORMAT = "dd-M-yyyy";
+    private static final String TIME_ZONE = "Asia/Jerusalem";
+    private SimpleDateFormat dateAndTimeFormatter;
```

Comment:
If `dateAndTimeFormatter` is only dependent on `DATE_TIME_FORMAT` etc, then you can also make it `static final` (and also UPPER_SNAKE_CASE).

You can create a static method to set both format and timezone.

Reply:
Ok thanks! Fixed it, but I had to initialize them on deceleration stage as they are finals. Correct?
About the time zones, created initTimeZones() for setting that. I'm not sure how much it values in term of code shortening, but it makes onCreate() much neater IMO. Is it worth it?

Removed `DATE_TIME_FORMAT` and `DATE_FORMAT` to be part of SimpleDateFormat construction. As part of trying to reduce unnecessary variables.